### DISCUSSION TITLE
关于粘滞会话(Sticky Session)的请求方实现机制

### DISCUSSION THREAD

**Discussion by songshuohua** [alibaba/alibaba-rsocket-broker#225] [Q&A] [answered]
阅读 alibaba-rsocket-broker 源码时发现，RSocketBrokerResponderHandler 会记录该 requester 首次请求某个服务时，当前 broker 选定的 provider，以此实现 broker 端的粘性会话。但客应用端 UpstreamCluster 存在负载均衡机制，如果 requester 第二次请求同一服务，通过负载均衡选定了不同的 broker 来转发请求，此时如何保证粘性会话依然有效，望大佬 @linux-china 解惑

**Comment by linux-china** [comment, marked as answer]
详细的介绍在这里： https://github.com/alibaba/alibaba-rsocket-broker/wiki/RSocket-Routing   这个有一定的局限性，如负载均衡问题，不能完全做到，这个要注意一下。

**Reply by songshuohua** [reply]
多谢大佬！
